### ISSUE TITLE
Open Interest Bybit

### ISSUE THREAD

**Issue by Sebastian7700** [AtasPlatform/Indicators#2]
As written here, the open interest indicator currently only works in the Russian markets.
https://support.atas.net/de/knowledge-bases/2/articles/8560-open-interest-o

Since Open Interest is extremely useful in trading, it would be really great to also have it in other markets. I can't speak for all markets, but Bybit is providing an API from which one can easily get the open interest Data.
It would be great if the indicator could display this data. You can find the documentation here:
https://bybit-exchange.github.io/docs/inverse/?console#t-marketopeninterest

Best regards and thanks in advance!

**Comment by CCefman** [comment]
Hi hope  you're well. Rather then opening a new issue I thought I'd add a comment to this one instead. Not sure if this will get anyone's attention considering it's already closed but ill keep checking the develop branch to see if you implemented this. 

First of all I've never try to write a script myself so I do apologise if I don't make the most sense or if I'm asking for something which is way too much, but this would really really useful!
![OI Example 1](https://user-images.githubusercontent.com/92552626/141493566-7220df37-c217-43d6-a50d-efeff0d331d6.png)

I've noticed that currently there are very limited statistics that you can see on Open Interest. I was hoping you'd be able to add in OI Delta statistics so you could see it above/under every candle. Check Attachment 1 to get a better idea of what I mean. 

Further Improving to this would be adding some form of ZigZag Statistics so you can see the % change of the move rather then on each candle. Ideally if you could add "Minimum ZigZag Leg %" so we can adjust that ourselves that would be incredible. Have a look at Attachment 2 to get a better idea.
![OI Example 2](https://user-images.githubusercontent.com/92552626/141494105-d84a030d-cc03-452a-8d71-95327e70fdec.png)
 
Again I apologise if this is way too much to ask but as open interest is incredibly important in trading this would be a very useful improvement. Thanks a lot again, and ill keep checking the develop branch.

**Comment by Sebastian7700** [comment]
Greetings,
I reopened this issue since @CCefman had a really good idea for further improvement. Although before this can be added, the indicator has to work. @ErtyHackward  you wrote that you fixed the problem with the open interest indicator. I don't know if this is a problem on my end, but for me, the indicator prints one candle when you open ATAS, and then it stops working. Also, it seems to be displaying the wrong value. I tried downloading ATAS 5.7.26 beta, but no difference.
![image](https://user-images.githubusercontent.com/61465142/141504269-55a3edf0-7c25-41d4-990a-f7953ad7b7b8.png)

I've also tried to get some data from the Bybit API, all be it with Python.
First of all, to my knowledge, there are two API calls that will return the open interest.

- A. curl https://api.bybit.com/v2/public/open-interest
with this, I have been able to get OI data for specific timeframes up to 200 data points back.
I don't know if you have access to other timeframes, but I think publicly only the ones that are listed here are available:
https://bybit-exchange.github.io/docs/inverse/?console#t-marketopeninterest
What I have found is that this call returns the data points in a different order than the call for price data, e.g.:
in price data, it's ordered: past -> present
in oi data, it's ordered: present -> past
I thought if you are using this API call, this could be part of the problem.
![image](https://user-images.githubusercontent.com/61465142/141507945-013d5755-d17d-4ede-906f-6af1136ae07a.png)
This is what I was able to get with a little Python script. (Tell me if you want to have a look at it.)

- B. curl https://api.bybit.com/v2/public/tickers
this will return all current data for all the symbols on Bybit, also open interest data.
This will only return the current OI value, unlike the first call, where you could specify how many data points you want. You can find it here:
https://bybit-exchange.github.io/docs/inverse/?console#t-latestsymbolinfo

The value from B. and the most present value from A. are not exactly the same, I think it is because A. only gives finished candles and B. gives the exact present value.
While writing this they return about 320 Million, while the one data bar from ATAS shows about 190 Million.

Although I am not even sure if the 320 Million are correct since if you go to the Bybit website it shows about 2 Billion:
![image](https://user-images.githubusercontent.com/61465142/141509940-05de89b4-30c7-4af4-88ae-3a9fec12fa0c.png)

Thanks for your work, hope this helps you improve the platform!
Best regards Sebastian

P.S.
@CCefman are you by chance a member of ChartChampions?

**Comment by CCefman** [comment]
Wow @Sebastian7700 you’ve done a lot of digging into this issue from the looks of it. Very happy that you’ve reopened the issue with you further findings. Just want to confirm that I’ve also downloaded the new update and still the same problem with OI persists. I thought the fix just hasn’t been implemented yet but will be coming soon. 

And yes I joined them around a year ago never looked back since.

**Comment by ErtyHackward** [comment]
@CCefman Thanks for posting. I will check what we can do. You have a good proposal. Can you move it to a separate issue please? I think it is not directly related to this one and will be done by other guy.

@Sebastian7700 It's a shame for me but we didn't yet released the feature to the servers because of a bug that I fixed only last wednesday. 

Just for information, we're receving OI data from the [WebSocket endpoint](https://bybit-exchange.github.io/docs/inverse/?console#t-websocketinstrumentinfo). 

I will not give you an another estimate rather I post here when it will be done. I hope it will come soon.

**Comment by CCefman** [comment]
@ErtyHackward Thank for the quick response, really appreciate it. I have followed your suggestion and have linked my comment to a "New Issue". I understand that it will probably take some time to bring this to production environment but nevertheless this will be worth the wait 100%. Thanks again for your hard work and thank you @Sebastian7700 for going above and beyond trying to solve OI issue.

**Comment by Sebastian7700** [comment]
Hey there, over the last few days I've noticed that open interest is now working! That's awesome, thanks a lot! I noticed that at the moment it's only working on the F2 server, but I guess you start by testing it out on one server before pushing it to every server.
Not sure if you are aware but sometimes, some servers are giving delayed data. E.g. today the newest data I was getting was from 20:00 but it was already 20:20. I tried "Refresh", "Reload all historical data" and restarting ATAS but it didn't change. Only after changing from the F2 server to the Auto option did I receive live data.
